src/main/java/com/github/tomakehurst/wiremock/extension/responsetemplating/helpers/JoinHelper.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2019-2024 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.extension.responsetemplating.helpers;
+
+import com.github.jknack.handlebars.Options;
+import com.github.jknack.handlebars.TagType;
+import java.io.IOException;
+import java.util.List;
+
+public class JoinHelper extends HandlebarsHelper<Object> {
+
+  @Override
+  public Object apply(Object context, Options options) throws IOException {
+
+    if (context == null || !(context instanceof String)) {
+      return handleError("Separator parameter must be a String");
+    }
+
+    String separator = (String) context;
+
+    Object param = options.param(0, null);
+    if (param == null || !(Iterable.class.isAssignableFrom(param.getClass()))) {
+      return handleError("The parameter must be list");
+    }
+    List<Object> items = (List<Object>) param;
+    if (items.isEmpty()) {
+      return "";
+    }
+    String result;
+    if (options.tagType == TagType.SECTION) {
+      result = processSection(options, separator, items);
+    } else {
+      result = processWithoutSection(separator, items);
+    }
+
+    return result.replaceAll("\n", "");
```

Comment:
I wasn't sure what the purpose of this was.  Is there a reason we wouldn't want to let people have new line characters in their output?  For example, this test would fail and I am not sure why it is an issue? 

```java
  @Test
  public void joinWithArrayOfStringsWithNewLineCharacters() {
    String result = transform("{{join ',' (array 'One\n' 'Two\n' 'Three\n')}}");

    assertThat(result, equalToCompressingWhiteSpace("One\n,Two\n,Three\n"));
  }
```

Reply:
On the other hand, if we do this:
```
{{#join ',' myData |as| item}}
{
  "name": "{{item.name}}"
}
{{/join}}
```

We would like to obtain this `{name2:One},{name2:Two},{name2:Three}` , but instead, we obtain this:
```
{
name2:One
}
,
{
name2:Two
}
,
{
name2:Three
}
```

Reply:
Actually, for some reason I was assuming it should be in one line, but checking what `each` does, it prints what it's in there without removing anything, printing it in multiple lines. So, removing it 😃